app/actions/agentrelationship/request/AppointAgentSessionRequest.scala

```diff
@@ -41,4 +41,9 @@ case class AppointAgentSessionRequest[A](
   }
 
   def organisationId = groupAccount.id
+
+  def agentDetails = AgentDetails(
+    name = sessionData.agentOrganisationName.getOrElse(throw new IllegalStateException("no agent name stored")),
+    address = sessionData.agentAddress.getOrElse(throw new IllegalStateException("no agent address stored"))
+  )
```

Comment:
Feels like we should have a CcaAgentWrappedRequest or something to avoid these exceptions?